pkgs/tools/misc/git-town/default.nix

```diff
@@ -1,16 +1,25 @@
-{ lib, buildGoModule, fetchFromGitHub, installShellFiles, git, testers, git-town, makeWrapper }:
+{ lib, buildGoModule, fetchFromGitHub, fetchpatch, installShellFiles, git, testers, git-town, makeWrapper }:
 
 buildGoModule rec {
   pname = "git-town";
-  version = "7.7.0";
+  version = "7.8.0";
 
   src = fetchFromGitHub {
     owner = "git-town";
     repo = "git-town";
     rev = "v${version}";
-    sha256 = "sha256-FpBEBx2gb33fGDndvZmvG1A61NoJ4Qy4V3YQSb+Ugsc=";
+    sha256 = "sha256-g9ooMIMN8DN2FcWYkDC1hICCleQYdHf30PYMCit/NMI=";
   };
 
+  patches = [
+    # Fix "go vet" when building using Go 1.18.
+    (fetchpatch {
+      name = "fix-go-vet-in-go-1.18.patch";
+      url = "https://github.com/git-town/git-town/commit/23eb0aca7b28c6a0afc21db553aa0e35d35891aa.patch";
+      sha256 = "sha256-EyfhKVrQxRJNrYqaZI04dJogaXs1J+bbOIu7p8g2Clc=";
+    })
+  ];
```

Comment:
This should be disabled, we never want to run linters when building.

Reply:
I personally don't agree, unless the linter in question is purely focused on style enforcement. Even then, I would only diverge from upstream and disable it if it is causing other problems.

If this is a policy within `nixpkgs`, I don't mind following it, however.

Reply:
> Even then, I would only diverge from upstream and disable it if it is causing other problems.

If it is causing problems it is already extra maintenance burden we could have avoided.

At least for python packages we enforce disabling linters and I think we should follow that in other places, too.

Reply:
We are optimizing for different things. Your opinion is legitimate too, no hard feelings. I'm currently not interested in pursuing disabling linting for Go projects. If it's a blocker, I'll close this PR.

Reply:
There was no reason to close the PR and delete the branch without even waiting for a response.

> This shouldn't matter. Are we running go vet as part of the build process?

`go test` runs some `go vet` checks. If you are of the opinion that it would be better to disable them, it can be enforced in `buildGoModule`/`buildGoPackage` with `-vet=off`. I see no point in disabling them in only some individual packages.

Having this patch is perfectly fine, it's like any other upstream patch that we fetch 
while waiting for the next release.

Reply:
I restored the branch and reopened the PR. I can keep it open for discussion.

Reply:
Can we merge this PR? I think the discussion about what should and should not be run when running go test is out of scope here.

Reply:
> `go test` runs some `go vet` checks. If you are of the opinion that it would be better to disable them, it can be enforced in `buildGoModule`/`buildGoPackage` with `-vet=off`. I see no point in disabling them for only some individual packages.

Thanks, creating a PR to staging right now to fix similar issues in the future.